static/app/views/settings/organizationDeveloperSettings/sentryApplicationDetails.tsx

```diff
@@ -343,26 +344,24 @@ class SentryApplicationDetails extends DeprecatedAsyncComponent<Props, State> {
     }
   };
 
-  onFieldChange = (name: string, value: FieldValue): void => {
-    if (name === 'webhookUrl' && !value && this.isInternal) {
+  const onFieldChange = (name: string, value: FieldValue): void => {
+    if (name === 'webhookUrl' && !value && isInternal()) {
       // if no webhook, then set isAlertable to false
-      this.form.setValue('isAlertable', false);
+      form.setValue('isAlertable', false);
     }
   };
 
-  addAvatar = ({avatar}: Model) => {
-    const {app} = this.state;
+  const addAvatar = ({avatar}: Model) => {
     if (app && avatar) {
       const avatars =
         app?.avatars?.filter(prevAvatar => prevAvatar.color !== avatar.color) || [];
 
       avatars.push(avatar as SentryAppAvatar);
-      this.setState({app: {...app, avatars}});
+      setApiQueryData(queryClient, SENTRY_APP_QUERY_KEY, {...app, avatars});
```

Comment:
Q so does `useApiQuery` also serve as a `useState` for the parameters/data we extracted from the api call/`useApiQuery` does both api call + state management? 

Then now when we're setting/updating the data from the api call, does that make another api call ? Or just do the state management piece/behave like the setter in `useState`

Reply:
I believe `useApiQuery` also serves as a `useState`. It doesn't make another API call, according to the [docs](https://tanstack.com/query/v5/docs/reference/QueryClient#queryclientsetquerydata) `setApiQueryData` just updates the cached value

I used the account details form as an example 😳 
https://github.com/getsentry/sentry/blob/0cfd30073ce86d52c54e366b0fc945511d0cc973/static/app/views/settings/account/accountDetails.tsx#L65-L72